.github/workflows/code-hygiene.yml

```
@@ -0,0 +1,11 @@
+name: Code Hygiene
+on: [pull_request]
+jobs:
+  analyze:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: astral-sh/uv-action@v1
+      - run: uv run python scripts/analyze_codebase.py --out CODE_HEALTH_DUPLICATES_DEAD_CODE.md --json report.json
+      - run: |
+          jq '.duplicates.files + .duplicates.blocks + .unreferenced.count' report.json | awk '{exit ($1>0)}'
```

YAMLlint (1.37.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)